### DISCUSSION TITLE
How to have green tunnel run on startup? (Debian)

### DISCUSSION THREAD

**Discussion by Shane8johnson** [SadeghHayeri/GreenTunnel#148] [Q&A]
Can someone please assist/tell me how you would have green tunnel run on startup on the machine. I have added the command to the /etc/rc.local file, but it does not seem to actually start the proxy. Please help!

**Comment by Charlie284** [comment]
`/etc/rc.local` is a poor fit for this because it runs with a minimal boot environment: it normally will not source your shell profile or NVM setup, and GreenTunnel's Linux system-proxy integration uses GSettings, which belongs to a logged-in desktop session.

For the current GreenTunnel CLI, I would run the proxy as your normal user with a systemd user service and configure clients explicitly.

First find the real installed paths:

```sh
dirname "$(command -v node)"
command -v gt
```

Create `~/.config/systemd/user/green-tunnel.service` (create the directory if needed):

```ini
[Unit]
Description=GreenTunnel local proxy

[Service]
Type=simple
# Replace both paths with the output of the commands above. If Node came from
# NVM, keep that version's bin directory here because systemd does not load
# ~/.profile or ~/.bashrc.
Environment=PATH=/absolute/path/to/node/bin:/usr/local/bin:/usr/bin:/bin
ExecStart=/absolute/path/to/gt --host 127.0.0.1 --port 8000 --no-system-proxy --log-level info
Restart=on-failure
RestartSec=5

[Install]
WantedBy=default.target
```

Then load and start it:

```sh
systemctl --user daemon-reload
systemctl --user enable --now green-tunnel.service
systemctl --user status green-tunnel.service
journalctl --user -u green-tunnel.service -f
```

That starts it when your user manager starts. If it must start during boot before you log in, enable lingering once:

```sh
sudo loginctl enable-linger "$USER"
```

Point the browser/application proxy at `127.0.0.1:8000`. Terminal programs do not use the desktop proxy automatically; set their environment where needed, for example:

```sh
export http_proxy=http://127.0.0.1:8000
export https_proxy=http://127.0.0.1:8000
```

The important details are the absolute `gt` path, an explicit Node `PATH`, and `--no-system-proxy`. Current GreenTunnel documents the [installation and CLI options](https://github.com/SadeghHayeri/GreenTunnel/blob/8278c2953e396c07af5ee4c1474fc183398e87a1/README.md#L66-L78), including the default `127.0.0.1:8000` endpoint and `--no-system-proxy` [here](https://github.com/SadeghHayeri/GreenTunnel/blob/8278c2953e396c07af5ee4c1474fc183398e87a1/README.md#L97-L127). Its Linux notes explain that proxy management is GSettings-based and terminal tools need `http_proxy`/`https_proxy` [here](https://github.com/SadeghHayeri/GreenTunnel/blob/8278c2953e396c07af5ee4c1474fc183398e87a1/README.md#L244-L251).

If Docker is already installed, the project's documented restart policy is a simpler alternative. Bind only to loopback unless you deliberately intend to share the proxy on the LAN:

```sh
docker run -d --name green-tunnel --restart unless-stopped \
  -p 127.0.0.1:8000:8000 sadeghhayeri/green-tunnel
```

The original 2022 release used `--ip` rather than today's `--host`, so if you are intentionally staying on that old version, adjust the argument. Updating is preferable because the current CLI and runtime requirements have changed substantially.
